openpype/hosts/houdini/plugins/publish/collect_cache_farm.py

```diff
@@ -0,0 +1,87 @@
+import os
+import pyblish.api
+import hou
+from openpype.hosts.houdini.api import lib, colorspace
+from openpype.hosts.houdini.api.lib import get_color_management_preferences
+
+
+class CollectDataforCache(pyblish.api.InstancePlugin):
+    """Collect data for caching to Deadline."""
+
+    order = pyblish.api.CollectorOrder + 0.04
+    families = ["ass", "pointcache",
+                "mantraifd", "redshiftproxy"]
+    hosts = ["houdini"]
+    targets = ["local", "remote"]
+    label = "Collect Data for Cache"
+    chunkSize = 999999
+
+    def process(self, instance):
+        if instance.data.get("farm", True):
+            for family in instance.data["families"]:
+                instance.data["family"] = family
+                if family == "bgeo" or "abc":
+                    instance.data["family"] = "pointcache"
+                instance.data["subset"] = instance.name
+            # instance.data["expected_families"]
+            # only created for farm caching
+            instance.data["expected_families"] = list()
+
+            instance.data["expected_families"].append(
+                instance.data["family"])
+            instance.data["families"].append("publish.hou")
+            data = dict()
+
+            data["plugin"] = "Houdini"
+            data["publish"] = True
+            ropnode = hou.node(instance.data["instance_node"])
+            instance.data.update(data)
+            output_parm = lib.get_output_parameter(ropnode)
+            # TODO: make sure the expectedfiles include all the data
+            expected_filepath = output_parm.eval()
+            if "files" not in instance.data:
+                instance.data["files"] = list()
+            if "expectedFiles" not in instance.data:
+                instance.data["expectedFiles"] = list()
+            cache_files = dict()
+            if instance.data.get("frames"):
+                files = self.get_files(
+                    instance, expected_filepath)
+                # list of files
+                instance.data["files"].extend(files)
+            else:
+                instance.data["files"].append(output_parm.eval())
+            cache_files = {"_": instance.data["files"]}
+            instance.data["expectedFiles"].append(cache_files)
+
+            # update the colorspace data
+            colorspace_data = get_color_management_preferences()
+            instance.data["renderProducts"] = colorspace.ARenderProduct()
+            instance.data["colorspaceConfig"] = colorspace_data["config"]
+            instance.data["colorspaceDisplay"] = colorspace_data["display"]
+            instance.data["colorspaceView"] = colorspace_data["view"]
+            self.log.debug("{}".format(instance.data))
+
+        else:
+            self.log.debug("Caching at farm disabled!")
```

Comment:
Here's basically this logic re-written somewhat (consider it pseudocode) but with some question comments added in there. There's a lot going on here that's no apparent at first sight WHY it's needed - so it'd be good to put some comments in to explain why certain things are needed. E.g. 
- Why do we need expectedFamilies? Why can't we rely on the instance's families? 
- Why are we remapping subsets? 
- That colorspace data being collected for the instance doesn't seem very "renderfarm" related. Why can't there just be a a single `CollectColorspaceConfig` which collects it maybe to the context? (collects it once instead of per instance), etc.

```suggestion
        if not instance.data.get("farm", True):
            self.log.debug("Caching on farm is disabled. "
                           "Skipping farm collecting.")
            return

        # Why do we need this particular collector to collect the expected
        # output files from a ROP node. Don't we have a dedicated collector
        # for that yet?
        # Collect expected files
        ropnode = hou.node(instance.data["instance_node"])
        output_parm = lib.get_output_parameter(ropnode)
        expected_filepath = output_parm.eval()

        instance.data.setdefault("files", list())
        instance.data.setdefault("expectedFiles", list())
        if instance.data.get("frames"):
            files = self.get_files(instance, expected_filepath)
            # list of files
            instance.data["files"].extend(files)
        else:
            # single file
            instance.data["files"].append(output_parm.eval())
        cache_files = {"_": instance.data["files"]}

        # Convert instance family to pointcache if it is bgeo or abc
        # because ???
        for family in instance.data["families"]:
            if family == "bgeo" or "abc":
                instance.data["family"] = "pointcache"
                break

        instance.data.update({
            "expectedFamilies": [instance.data["family"]],

            # Why are we overriding the subset?
            "subset": instance.name,

            # Why do we need these in `instance.data`?
            "plugin": "Houdini",
            "publish": True
        })
        instance.data["families"].append("publish.hou")
        instance.data["expectedFiles"].append(cache_files)

        # update the colorspace data
        colorspace_data = get_color_management_preferences()
        instance.data["renderProducts"] = colorspace.ARenderProduct()
        instance.data["colorspaceConfig"] = colorspace_data["config"]
        instance.data["colorspaceDisplay"] = colorspace_data["display"]
        instance.data["colorspaceView"] = colorspace_data["view"]
        self.log.debug("{}".format(instance.data))
```